client/display/style.rs

```diff
@@ -0,0 +1,100 @@
+use pueue_lib::settings::Settings;
+
+use comfy_table::Cell;
+use crossterm::style::{style, Attribute, Color, Stylize};
+
+/// OutputStyle wrapper for actual colors depending on settings
+/// - Enables styles if color mode is 'always', or if color mode is 'auto' and output is a tty.
+/// - Using dark colors if dark_mode is enabled
+pub struct OutputStyle {
+    /// whether or not ANSI styling is enabled
+    enabled: bool,
+    /// red color
+    red: Color,
+    /// green color
+    green: Color,
+    /// yellow color
+    yellow: Color,
+}
+
+impl OutputStyle {
+    /// init color-scheme depending on settings
+    pub const fn new(settings: &Settings, enabled: bool) -> Self {
+        if settings.client.dark_mode {
+            Self {
+                enabled,
+                green: Color::DarkGreen,
+                red: Color::DarkRed,
+                yellow: Color::DarkYellow,
+            }
+        } else {
+            Self {
+                enabled,
+                green: Color::Green,
+                red: Color::Red,
+                yellow: Color::Yellow,
+            }
+        }
+    }
+
+    /// This is a helper method with the purpose of easily styling text,
+    /// while also prevent styling if we're printing to a non-tty output.
+    /// If there's any kind of styling in the code, it should be done with the help of this method.
+    pub fn style_text<T: ToString>(
+        &self,
+        text: T,
+        color: Option<Color>,
+        attribute: Option<Attribute>,
+    ) -> String {
+        let text = text.to_string();
+        // Styling disabled
+        if !self.enabled {
+            return text;
+        }
+
+        let mut styled = style(text);
+        if let Some(color) = color {
+            styled = styled.with(match color {
+                Color::Green => self.green,
+                Color::Red => self.red,
+                Color::Yellow => self.yellow,
+                _ => color,
+            });
```

Comment:
You could extract this duplicate match block into a separate `self.color` or `self.shade` function.

Reply:
Excelent idea!